src/coord/src/catalog.rs

```diff
@@ -1086,6 +1087,8 @@ impl Catalog {
         } else {
             schema.items.insert(entry.name.item.clone(), entry.id);
         }
+        metrics::item_created(entry.id, &item);
```

Comment:
I think it would be cleaner to call `item_created` and `item_dropped` in [`process_catalog_events`](https://github.com/MaterializeInc/materialize/blob/7e771517817b6449a23801428a0adf87b48ff6d8/src/coord/src/coord.rs#L2791). `process_catalog_events` is always called after catalog items are created, updated, and dropped, and it's where we maintain `mz_*` table information currently.

Reply:
That's a great call! I wholeheartedly endorse @JLDLaughlin's suggestion.

Reply:
Ah, yeah I was a little torn between these two places, done.